### PR TITLE
Re:Cargo

### PR DESCRIPTION
**Pull request by AlexMorgan3817**
## About The Pull Request

- [x] Cargo rework itselfs
<hr>

- [x] NanoUI template for trade program
- [x] Adding of default trade stations
- [x] Add more specific trade stations
- [x] Trade station can have "forced" position
- [x] Custom categories of goods for trade station (need fix)
- [x] Add way to get trade program, without NTNet download, or just put disk with it to guild.

## Changelog
:cl:
refactor: Cargo system reworked, now it is trade system, using overmap.
/:cl:

### CODE REVIEW

**Comment by ArmolitskiyCarpoeb** [comment]
nice

File: code/modules/trade/datums/trade_stations_presets/caduceus.dm [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/datum/trade_station/caduceus
+	name_pool = list("SAV 'Caduceus'" = "old Sol republic Aid Vessel 'Caduceus'. They sending message, \"Hello, there, we are from Old Sol Republic, we will soon jump back, cuz we get what we wanted, and while we here we can offer medical help for you.\"")
```
Comment (GrayRachnid) [inline code/modules/trade/datums/trade_stations_presets/caduceus.dm:2]: They're sending a message. Hello there, we are from the Old Sol Republic. We will be leaving the system shortly but we can offer you any medical aid while we are still here.

File: code/modules/trade/datums/trade_stations_presets/hellcat.dm [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/datum/trade_station/fbv_hellcat
+	name_pool = list("FBV 'Hellcat'" = "They are sending message, \"Hello, we escorting SAV 'Caduceus', but while we here we can offer you some our extra supplies.\"")
```
Comment (GrayRachnid) [inline code/modules/trade/datums/trade_stations_presets/hellcat.dm:2]: Greetings. This is the Hellcat. We're currently escorting the Caduceus and we will be departing the system shortly alongside them. We are willing to depart with some extra supplies to get rid of while we're still here.

File: code/modules/trade/datums/trade_stations_presets/zarya.dm [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/datum/trade_station/station_zarya
+	name_pool = list("FTS 'Zarya'" = "Free Trade Station 'Zarya', they sending message \"Privet, kosmicheskiy strannik, u nas mnogo engenernih nishtyakov, zaletai!\".")
```
Comment (GrayRachnid) [inline code/modules/trade/datums/trade_stations_presets/zarya.dm:2]: )))))))
Privet, you're speaking to the free station of Zarya. We have everything for sale here, don't be afraid to come aboard and check our wares!

File: code/modules/stashes/stash_types/weapons.dm [hunk 4]
```diff
@@ -101,15 +102,15 @@
...
+
 /datum/stash/weapon/mutiny/warden
 	contents_list_extra = list(/obj/random/gun_normal = 3, /obj/random/ammo = 6)
 	directions = DIRECTION_IMAGE
 	lore = "Warden,<br>\
  You and I both know this rubber buckshot bullshit is going to get us all killed when the convicts realize they outnumber us sixty to one.<br>\
  Sarish tells me there's a stash of real heaters wherever the hell this is.<br>\
  Crazy bitch sends a picture instead of a fucking coordinate. %D"
-*/
+
```
Comment (Kurgis) [inline code/modules/stashes/stash_types/weapons.dm:113]: Not sure why these changes were undone, is this up to date with master?

Comment (AlexMorgan3817) [reply code/modules/stashes/stash_types/weapons.dm:113]: I don't know, i think it was here when i got root's cargo rework

Comment (Kurgis) [reply code/modules/stashes/stash_types/weapons.dm:113]: Undo these changes when able, since that datum is broken.

**Comment by AlexMorgan3817** [comment]
Almost ready for review, waiting for my in-game map test

File: code/__DEFINES/misc.dm [hunk 5]
```diff
@@ -283,13 +283,11 @@
...
-#define FOURSPACES "&nbsp;&nbsp;&nbsp;&nbsp;"
-
 #define NEWorINITIAL(variable, newvalue) variable = newvalue ? newvalue : initial(variable)
 
 //Matricies
 #define MATRIX_GREYSCALE list(0.33, 0.33, 0.33,\
                               0.33, 0.33, 0.33,\
                               0.33, 0.33, 0.33)
+
+#define VAR_OUT_TO_TEXT(var) "[#v] = [v]"
```
Comment (TheShown911) [inline code/__DEFINES/misc.dm:293]: No newline.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 6]
```diff
@@ -1655,7 +1655,7 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/bigbiteburger
 	name = "Big Bite Burger"
-	desc = "Forget the Big Mac. THIS is the future!"
+	desc = "Forget the Big Mac. THIS is the future! It have big \"R\" stamp on it's bun."
```
Comment (TheShown911) [inline code/modules/reagents/reagent_containers/food/snacks.dm:1658]: Should be "It has a big \"R\" stamped on it's bun. "

File: code/modules/trade/datums/_trade_station.dm [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+	var/turf/overmap_location
+
+/datum/trade_station/New(init_on_new)
+	. = ..()
+	if(init_on_new)
+		init_src()
+
+/datum/trade_station/proc/init_src()
+	if(name)
+		crash_with("Some retard gived trade station a name before init_src, not thought name_pool. ([type])")
```
Comment (TheShown911) [inline code/modules/trade/datums/_trade_station.dm:40]: This is not english, though spooky broken english is more effective at motivating someone if they make a mistake.

**Review by drexample — APPROVED** [review]

File: code/modules/trade/datums/trade_stations_presets/cybernetics.dm [hunk 8]
```diff
@@ -0,0 +1,29 @@
+/datum/trade_station/cybermoebus
+	name_pool = list("FTS 'PentaOptium'" = "Free Trade Station 'PentaOptium'. \"Oh, you, i think i can sold you some cybernetics.\"")
```
Comment (TheShown911) [inline code/modules/trade/datums/trade_stations_presets/cybernetics.dm:2]: Many descriptions don't make much sense and don't follow the english language exactly, but i have been informed that these bing **soul** into the game so i'll let them pass. Also they are very funny.

Comment (AlexMorgan3817) [reply code/modules/trade/datums/trade_stations_presets/cybernetics.dm:2]: If you want to help with that, offer normal description, not a "This descriptions is bad, fix it"... :c

File: code/controllers/subsystems/trade.dm [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+
+	var/points = 0
+
+	for(var/atom/movable/AM in beacon.get_objects())
+		if(AM.anchored)
+			continue
+
+		var/export_cost = get_export_cost(AM)
+		if(!export_cost)
+			return
```
Comment (AlexMorgan3817) [inline code/controllers/subsystems/trade.dm:130]:
```suggestion
			continue
```

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 6 above)
Comment (AlexMorgan3817) [inline code/modules/reagents/reagent_containers/food/snacks.dm:1658]:
```suggestion
	desc = "Forget the Big Mac. THIS is the future! It has big \"R\" stamped on it's bun."
```

File: code/modules/trade/datums/trade_stations_presets/cybernetics.dm (hunk 8 above)
Comment (TheShown911) [reply code/modules/trade/datums/trade_stations_presets/cybernetics.dm:2]: I genuinely like these descriptions, and am not very good with English myself. When I had bad descriptions in my PR I got help from en docks to correct them. Though a few grammatically incorrect descriptions don't ruin a PR.

**Comment by AlexMorgan3817** [comment]
Done

File: code/__DEFINES/misc.dm [hunk 10]
```diff
@@ -171,8 +171,8 @@
 #define ANNOUNSER_NAME "CEV Eris System Announcer"
 
 
-#define LIST_OF_CONSONANT list("b", "c", "d", "f", "g", "h", "j", "k", "l", "m", "n", "p", "q", "r", "s", "t", "v", "w", "x", "y", "z", "á", "â", "ã", "ä", "æ", "ç", "é", "ê", "ë", "ì", "í", "ï", "ð", "ñ", "ò", "ô", "õ", "ö", "÷", "ø", "ù")
-
+#define LIST_OF_CONSONANT list("a", "b", "c", "d", "f", "g", "h", "j", "k", "l", "m", "n", "p", "q", "r", "s", "t", "v", "w", "x", "y", "z", "á", "â", "ã", "ä", "æ", "ç", "é", "ê", "ë", "ì", "í", "ï", "ð", "ñ", "ò", "ô", "õ", "ö", "÷", "ø", "ù")
+#define EN_ALPHABET list("a", "b", "c", "d", "f", "g", "h", "j", "k", "l", "m", "n", "p", "q", "r", "s", "t", "v", "w", "x", "y", "z")
```
[the remark this answers is not in the document]
Comment (TorinoFermic) [reply code/__DEFINES/misc.dm:175]: Also, vowels are "a", "e", "o" "i", "u" and won't be in consonants

**Review by TheShown911 — APPROVED** [review]
